aws/sdk/examples/s3-helloworld/src/main.rs

```diff
@@ -0,0 +1,40 @@
+use s3::Region;
+use smithy_http::byte_stream::ByteStream;
+use std::error::Error;
+use std::path::Path;
+use tracing_subscriber::fmt::format::FmtSpan;
+use tracing_subscriber::fmt::SubscriberBuilder;
+
+// Change these to your bucket & key
+const BUCKET: &str = "demo-bucket";
+const KEY: &str = "demo-object";
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error>> {
+    SubscriberBuilder::default()
+        .with_env_filter("info")
+        .with_span_events(FmtSpan::CLOSE)
+        .init();
+    let conf = s3::Config::builder()
+        .region(Region::new("us-east-2"))
+        .build();
+    let client = s3::Client::from_conf(conf);
+    let resp = client.list_buckets().send().await?;
+    for bucket in resp.buckets.unwrap_or_default() {
+        println!("bucket: {:?}", bucket.name.expect("buckets have names"))
+    }
+    let body = ByteStream::from_path(Path::new("Cargo.toml")).await?;
```

Comment:
Not blocking: should `ByteStream::from_path` be generic over `<P: AsRef<Path>>` so that customers can pass a `&str` in directly rather than constructing a path?

Reply:
that's a good question. I think it probably should be. Originally it wasn't because `ByteStream::from(&str)` is obviously very different. But given that the function is named `from_path` (this was required because it ended up being async) we can accept a more lenient argument 👍🏻